components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountDisableHandler.java

```diff
@@ -386,6 +408,48 @@ private void setUserClaim(String claimName, String claimValue, UserStoreManager
         }
     }
 
+    private void publishPreAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map) throws
+            AccountLockException {
+
+        AccountUtil.publishEvent(accountDisabledEventName, AccountUtil.cloneMap(map));
+    }
+
+    private void publishPostAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map, boolean
+            isDisblePropertySuccessfullyModified) throws AccountLockException {
+
+        Map<String, Object> eventProperties = AccountUtil.cloneMap(map);
```

Comment:
Why do we need to deep-copy the Map here?

Reply:
Because we shouldn't pass the same object as per the code review comment by @IsuraD